Pop-Grupa-A-Backend/app/main/services/task_service.py

```diff
@@ -1,68 +1,53 @@
 from flask import request, abort
 
-from app.main.model.ComputationTask import ComputationTask
-from app.main.model.ComputationStatus import ComputationStatus
-from app.main.model.ComputationAccount import ComputationAccount
-from app.main.model.ComputationApplication import ComputationApplication
-
 import app.main.repositories.task_repository as task_repository
-from ..repositories.user_repository import get_user 
+import app.main.repositories.user_repository as user_repository
+import app.main.repositories.application_repository as application_repository
 
 def add_task(task):
-    db_user = ComputationAccount.query.filter_by(id=task['user_id']).first()
-    db_app = ComputationApplication.query.filter_by(id=task['app_id']).first()
+    db_user = user_repository.get_user(task['user_id'])   
+    db_app =  application_repository.get_app_by_id(task['app_id'])
 
     if db_user and db_app:
-        new_task = ComputationTask(
-            status = ComputationStatus.SUBMITTED,
-            user_id = task['user_id'],
-            app_id = task['app_id']
-        )
-
-        task_repository.save_changes(new_task)
-
+        task_repository.add_new_task(task)
         return {
                 'status': 'Success',
                 'message': 'Task successfuly created.'
                 }, 201
     if not db_app:
         return {
-                'status': 'Failure',
+                'status': 'Fail',
                 'message': f"App with id = {task['app_id']} does not exist",
                 }, 400
-     
     return {
-            'status': 'Failure',
+            'status': 'Fail',
             'message': f"User with id = {task['user_id']} does not exist",
             }, 400
 
 
 def activate_task(task_id):
     try:
-        task = task_repository.get_task_for_task_id(task_id)
-        task.status=ComputationStatus.WORKING
-        task_repository.save_changes(task)
+        task = task_repository.change_task_status_to_active(task_id)
```

Comment:
"task =" chyba nie potrzebne, samo "task_repository.change_task_status_to_active(task_id)"wystarczy. Ogólnie wydaję mi się, że w repository powinny być bardziej generyczne metody np. można task_repository.change_task_status(task_id, ComputationStatus.WORKING), ale to jak chcesz.

Reply:
Tu masz racje poprawię